tests/crypto_policy_test.go

```diff
@@ -0,0 +1,240 @@
+		}
+	}
+	return
+}
+
+var (
+	old          = ocpv1.TLSSecurityProfile{Type: "Old", Old: &ocpv1.OldTLSProfile{}}
+	intermediate = ocpv1.TLSSecurityProfile{Type: "Intermediate", Intermediate: &ocpv1.IntermediateTLSProfile{}}
+	modern       = ocpv1.TLSSecurityProfile{Type: "Modern", Modern: &ocpv1.ModernTLSProfile{}}
+	custom       = ocpv1.TLSSecurityProfile{Type: "Custom",
+		Custom: &ocpv1.CustomTLSProfile{
+			TLSProfileSpec: ocpv1.TLSProfileSpec{
+				Ciphers:       []string{"TLS_AES_128_GCM_SHA256", "TLS_CHACHA20_POLY1305_SHA256"},
+				MinTLSVersion: ocpv1.VersionTLS13,
+			},
+		},
+	}
+
+	tls10AllowedCipher = "ECDHE-ECDSA-AES128-SHA"
+
+	// Note that "crypto/tls" does not support setting max tls version to anything below 1.2
+	oldPermutation tlsConfigTestPermutation = tlsConfigTestPermutation{
+		openshiftTLSPolicy: &old,
+		allowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS12,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+		disallowedConfigs: []clientTLSOptions{},
+	}
+
+	intermediatePermutation tlsConfigTestPermutation = tlsConfigTestPermutation{
+		openshiftTLSPolicy: &intermediate,
+		allowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS12,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+		disallowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS12,
+				OpenSSLCipherNames: []string{tls10AllowedCipher},
+			},
+		},
+	}
+
+	modernPermutation tlsConfigTestPermutation = tlsConfigTestPermutation{
+		openshiftTLSPolicy: &modern,
+		allowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS13,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+		disallowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS12,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+	}
+
+	customPermutation tlsConfigTestPermutation = tlsConfigTestPermutation{
+		openshiftTLSPolicy: &custom,
+		allowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS13,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+		disallowedConfigs: []clientTLSOptions{
+			{
+				MaxTLSVersion:      tls.VersionTLS12,
+				OpenSSLCipherNames: []string{},
+			},
+		},
+	}
+)
+
+func getPemCertificate() []byte {
+	var apiService apiregv1.APIService
+	err := apiClient.Get(ctx, client.ObjectKey{Name: "v1.build.openshift.io"}, &apiService)
+	Expect(err).ToNot(HaveOccurred())
+	pemCertificate := apiService.Spec.CABundle
+	return pemCertificate
+}
+
+func tryToAccessEndpoint(pod core.Pod, port, subpath string, tlsConfig clientTLSOptions) (attemptedUrl string, err error) {
+	portInt, err := strconv.ParseUint(port, 10, 16)
+	Expect(err).ToNot(HaveOccurred())
+	conn, err := portForwarder.Connect(&pod, uint16(portInt))
+	Expect(err).ToNot(HaveOccurred())
+	defer conn.Close()
+
+	certPool := x509.NewCertPool()
+	pemCert := getPemCertificate()
+	certPool.AppendCertsFromPEM(pemCert)
+
+	client := &http.Client{
```

Comment:
It there any difference between using this `client` and using `tls.Client()` function?
Would it be possible to use it, similarly to how it is used in `validator_test.go`?
https://github.com/kubevirt/ssp-operator/blob/b3a78fcf04bfe1cb0c21fbb5895138ee4e7bb603/tests/validator_test.go#L1167-L1182